.github/workflows/build_dev_docs.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: Deploy Development Documentation
+on:
+  push:
+    paths:
+      - 'docs/**'
+      - mkdocs.yml
+    branches:
+      - main
+jobs:
+  deploy:
+    name: Deploy Development Documentation
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: true
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Install Dependencies
+        run: |
+          pip install -r docs/requirements.txt
+      - name: Configure Git User
+        run: |
+          git config user.name "ci-docs"
+          git config user.email "ci-docs@gmail.com"
```

Comment:
Please change this to something more appropriate for us and include a comment that explains what is it.